src/bokeh/core/serialization.py

```diff
@@ -463,8 +463,10 @@ def _encode_other(self, obj: Any) -> AnyRep:
                 return None
 
         # handle array libraries that support conversion to a numpy array (e.g. polars, PyTorch)
-        if hasattr(obj, "__array__") and isinstance(arr := obj.__array__(), np.ndarray):
-            return self._encode_ndarray(arr)
+        if hasattr(obj, "__array__"):
+            warn(f"serializing {type(obj)} may result in copying data", BokehUserWarning)
```

Comment:
@aglebov I don't think we need to warn if using `__array__`, those results should generally be zero-copy *if possible*. When I mentioned warning, I had in mind a more general fallback that turned any iterable sequence into a list. A warning would definitely be advised for *that*.

Reply:
Sure, warning removed